src/components/search/data/searchLoader.js

```diff
@@ -1,5 +1,6 @@
+import { generatePath, redirect } from 'react-router-dom';
 import { getConfig } from '@edx/frontend-platform/config';
-import { ensureAuthenticatedUser } from '../../app/routes/data';
+import { ensureAuthenticatedUser } from '../../app/routes';
```

Comment:
You should be able to resolve the dependency cycle issue by keeping the removed `/data` from the previous import. Then, your mock in the test suite should change to reflect/match the updated import path here.

Checking out your PR locally, fixing the dependency cycle, and updating the mock's import path to include `/data` has tests passing.

```jsx
jest.mock('../../../app/routes/data', () => ({
  ...jest.requireActual('../../../app/routes/data'),
  ensureAuthenticatedUser: jest.fn(),
}));
```